packages/block-library/src/search/view.js

```diff
@@ -0,0 +1,53 @@
+window.addEventListener( 'DOMContentLoaded', () => {
+	const hiddenClass = 'wp-block-search__searchfield-hidden';
+
+	Array.from(
+		document.getElementsByClassName(
+			'wp-block-search__button-behavior-expand'
+		)
+	).forEach( ( block ) => {
+		const searchField = block.querySelector( '.wp-block-search__input' );
+		const searchButton = block.querySelector( '.wp-block-search__button' );
+		const activeElement = block.ownerDocument.activeElement;
+
+		const toggleSearchField = ( showSearchField ) => {
+			if ( showSearchField ) {
+				return block.classList.remove( hiddenClass );
+			}
+
+			return block.classList.add( hiddenClass );
```

Comment:
Nice suggestions! I applied them here: 3a5c81af6ce638b0c891f368b99e2098eecd4634

Reply:
In the closed state, the button is responsible for displaying the search form. In the open state, the button is responsible for _submitting_ the form. This is an important distinction, as we're changing the behavior of the button, and need to communicate that. So, I think we need to modify a couple of *the properties in the open state*:

- Remove `aria-expanded`: It's no longer controlling the hidden input, and having `aria-expanded` set makes screen readers announce it as `Search, expanded, button`, which makes it seem like pressing it again should collapse it instead of perform a search.
- Remove `aria-controls` for the same reason. It's no longer controlling the element.

When the search input is hidden, I agree with @mikachan that the search button should have a label to communicate that it will open the search form.

Thanks for working on this @jffng!

Reply:
Thanks @jeryj — should we keep `aria-expanded` and `aria-controls` when the search input is hidden, to accurately describe the button behavior? Or is that taken care of by adding an `aria-label` to the button?

Reply:
I applied this feedback in 919117f6a58f7b3ec9be83904dcd8172ea3f0018